### PR TITLE
Replace BbPromise usages

### PR DESCRIPTION
**Pull request by sleepwithcoffee**
Addresses: #7171

### CODE REVIEW

File: test/unit/lib/plugins/deploy.test.js [hunk 1]
```diff
@@ -25,11 +22,11 @@ describe('Deploy', () => {
   });
 
   describe('#constructor()', () => {
-    it('should have commands', () => expect(deploy.commands).to.be.not.empty);
-    it('should have hooks', () => expect(deploy.hooks).to.be.not.empty);
+    it('should have commands', () => expect(deploy.commands).not.be.empty);
+    it('should have hooks', () => expect(deploy.hooks).not.be.empty);
```
Comment (medikoo) [inline test/unit/lib/plugins/deploy.test.js:26]: Same here (and in other places)

File: test/unit/lib/plugins/deploy.test.js [hunk 2]
```diff
@@ -52,37 +49,35 @@ describe('Deploy', () => {
...
+      expect(spawnDeployFunctionStub.called).to.be.false;
+      expect(spawnPackageStub.calledOnce).to.be.true;
+      expect(spawnPackageStub.calledWithExactly('package')).to.be.true;
     });
 
     it('should throw an error if provider does not exist', () => {
       deploy.serverless.service.provider.name = 'nonExistentProvider';
 
-      return expect(deploy.hooks['before:deploy:deploy']()).to.be.rejectedWith(
+      return expect(deploy.hooks['before:deploy:deploy']()).to.eventually.be.rejectedWith(
```
Comment (medikoo) [inline test/unit/lib/plugins/deploy.test.js:80]: Let's also change function to _async_

**Review by medikoo — CHANGES_REQUESTED** [review]
Thanks @sleepwithcoffee for taking care of that!

In general it looks really good, I've just suggested few changes

**Comment by sleepwithcoffee** [comment]
for some reason, it's picking another branch, hmm
image

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**